test.cpp

```
@@ -0,0 +1,48 @@
+    Shape() {}
+    virtual void draw() {}
+    ~Shape() {}
+private:
+    int size;
+    int color;
+};
+
+class Circle : public Shape {
+public:
+    Circle() {}
+    void draw() { std::cout << "circle" << std::endl; }
```

Clang (14.0.6)

[error] annotate this function with 'override' or (rarely) 'final'

(modernize-use-override,-warnings-as-errors)